http-client-spi/src/main/java/software/amazon/awssdk/http/HttpMetric.java

```diff
@@ -104,6 +105,19 @@
     public static final SdkMetric<Integer> HTTP_STATUS_CODE =
         metric("HttpStatusCode", Integer.class, MetricLevel.TRACE);
 
+    /**
+     * The time taken to acquire a channel from the channel pool.
+     *
+     * <p>For HTTP/1 operations, a channel is equivalent to a TCP connection. For HTTP/2 operations, a channel is equivalent to
+     * an HTTP/2 stream channel. For both protocols, the time to acquire a new channel may or may not involve establishing a new
+     * connection, depending on whether an existing channel is available in the pool or not. New connection establishment time may
+     * also include TLS handshake time, if TLS is enabled.
+     *
+     * <p>Note: This metric is currently only supported in 'netty-nio-client'.
+     */
+    public static final SdkMetric<Duration> CHANNEL_ACQUIRE_DURATION =
+        metric("ChannelAcquireDuration", Duration.class, MetricLevel.INFO);
```

Comment:
Unsure if "ChannelAcquireDuration" or "ConnectionAcquireDuration" would be better. "Channel" may not be familiar terminology for some users, but "connection" may be slightly misleading for HTTP/2.

Reply:
This is a good point. For H2 I could imagine wanting to know both metrics. Is it possible to emit both physical socket acquire time, as well as stream channel acquire time as different metrics?

Reply:
`ConcurrencyAcquireDuration` sounds more consistent. Will use.

Socket acquire time should be possible. It may be a little complicated since the Netty ChannelPool is responsible for establishing a new connection. Would need to measure the time taken around the lowest-level pool, but only for when a new channel is created. I'll take a look as a follow-up to this PR.